src/Razor/test/Microsoft.AspNetCore.Razor.LanguageServer.Test/TestAdhocWorkspaceFactory.cs

```diff
@@ -0,0 +1,28 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.Host;
+
+namespace Microsoft.AspNetCore.Razor.LanguageServer.Test
+{
+    internal class TestAdhocWorkspaceFactory : AdhocWorkspaceFactory
+    {
+        public static readonly TestAdhocWorkspaceFactory Instance = new TestAdhocWorkspaceFactory();
```

Comment:
Would this be better to be `Instance => new TestAdhocWorkspaceFactory();` so instances aren't shared between tests? Or is it a good thing they're shared between tests?

I could see this going either way.

Reply:
So the workspace factory is a singleton and tecccchnically it shouldn't matter if it's shared.